schema_salad/python_codegen.py

```diff
@@ -88,17 +88,23 @@ def save(self):
         r = {}
 """)
 
-    def end_class(self, classname):
-        # type: (Text) -> None
+    def end_class(self, classname, field_names):
+        # type: (Text, List[Text]) -> None
 
         if self.current_class_is_abstract:
             return
 
         self.out.write("""
-        if errors:
-            raise ValidationException(\"Trying '%s'\\n\"+\"\\n\".join(errors))
-""" % self.safe_name(classname))
+        diff = set([str(k) for k in doc.keys()]) - set({attrs})
```

Comment:
You could optimize a little bit by initializing `attrs` as a class variable and using `frozenset()`, and then going through doc.keys() and testing to see if each key is in `self.attrs`.

Reply:
Thank you for the suggestion.

I changed `attrs` to a class variable.

Note that it is declared in the generated code by `end_class` not to introduce an extra parameter to `begin_class` and other methods.